### PR TITLE
Define per-AZ instancegroups for worker nodes

### PR DESCRIPTION
**Pull request by vijay-veeranki**
This is related to [2272](https://github.com/ministryofjustice/cloud-platform/issues/2272)

This will help to alter our kops config to control which AZ our worker nodes are created.

Also updated the recycle node job script to handle multiple AZ ig.

### CODE REVIEW

**Review by digitalronin — CHANGES_REQUESTED** [review]
Is there any way we can label the worker node instance groups with an additional label like "WORKER_NODES" or something?

It would be nice if we could come up with something where we don't need to keep editing the `recycle-node.rb` script every time we rename an instance group (e.g when we upgrade kubernetes).

If we had a "static" label in addition to the node group name, we could tell the script to pick a node to recycle from all the instancegroups which have that tag.

File: recycle-node.rb [hunk 1]
```diff
@@ -80,19 +80,19 @@ def node_ready?(node)
...
   node.dig("metadata", "labels")["kubernetes.io/role"] == "node" \
-    && node.dig("metadata", "labels", "kops.k8s.io/instancegroup") == WORKER_NODE_INSTANCEGROUP
+    && node.dig("metadata", "labels", "kops.k8s.io/instancegroup").include?(WORKER_NODE_INSTANCEGROUP)
 end
 
 def get_worker_instance_group_size
   docs = []
   YAML.load_stream(get_kops_config) { |doc| docs << doc }
-  worker_instance_group = docs.last
+  worker_instance_group = docs.last(3)
```
Comment (digitalronin) [inline recycle-node.rb:89]: Per my comment above, can we identify the worker node instance groups via a label?

If we can do that, then we don't need to worry about their position in the YAML file (i.e. `last(3)`). This is less brittle, so we won't break stuff if we reorder the sections of the yaml file.

File: recycle-node.rb [hunk 2]
```diff
@@ -80,19 +80,19 @@ def node_ready?(node)
 
 def worker_node?(node)
   node.dig("metadata", "labels")["kubernetes.io/role"] == "node" \
-    && node.dig("metadata", "labels", "kops.k8s.io/instancegroup") == WORKER_NODE_INSTANCEGROUP
+    && node.dig("metadata", "labels", "kops.k8s.io/instancegroup").include?(WORKER_NODE_INSTANCEGROUP)
 end
 
 def get_worker_instance_group_size
```
Comment (digitalronin) [inline recycle-node.rb:86]: We need to rename this method if it counts workers from multiple node groups. How about `target_worker_node_count` ?

**Review by digitalronin — CHANGES_REQUESTED** [review]

**Comment by vijay-veeranki** [comment]
Hi @digitalronin,

Updated the changes as suggested, can you please look in to it?

**Review by poornima-krishnasamy — DISMISSED** [review]

**Review by poornima-krishnasamy — APPROVED** [review]

**Pull request merged**